ballerina/tests/toXml_test.bal

```diff
@@ -1165,6 +1165,171 @@ isolated function testRecordWithNamespaceAnnotationToXml1() returns error? {
     test:assertEquals(result.toString(), expected, msg = "testComplexRecordToXml result incorrect");
 }
 
+type AddressR1 record {|
+    string city;
+    int code; 
+|};
+
+type Wsa_ReplyTo1 record {
+    @Namespace {prefix: "wsa", uri: "example1.com"}
+    AddressR1 Address;
+};
+
+type Htng_ReplyTo1 record {
+    @Namespace {prefix: "wsa", uri: "example1.com"}
+    AddressR1 Address;
+};
+
+@Name {value: "soap"}
+type Soap1 record {
+    @Name {value: "ReplyTo"}
+    @Namespace {prefix: "wsa", uri: "example1.com"}
+    Wsa_ReplyTo1 wsaReplyTo;
+    @Name {value: "ReplyTo"}
+    @Namespace {prefix: "htng", uri: "example2.com"}
+    Htng_ReplyTo1 htngReplyTo;
+};
+
+@test:Config {
+    groups: ["toXml", "testFail"]
+}
+isolated function testXmlToRecordWithNamespaceAttachedToFields() returns error? {
+    Soap1 val = {
+        htngReplyTo: {
+            Address: {
+                code: 40000,
```

Comment:
usually, this is a URI, not a physical address. :) But the example is ok.

Reply:
Just added it to check the nested case.